src/frontend/components/UI/QuestsViewer/components/QuestLogWrapper/index.tsx

```diff
@@ -9,38 +9,37 @@ import styles from './index.module.scss'
 import { useTranslation } from 'react-i18next'
 import useGetG7UserCredits from 'frontend/hooks/useGetG7UserCredits'
 import useGetPointsBalancesForProject from 'frontend/hooks/useGetPointsBalances'
-import useGetQuests from 'frontend/hooks/useGetQuests'
-import { useGetQuestStates } from 'frontend/hooks/useGetQuestStates'
+import { Quest } from '@hyperplay/utils'
 
 export interface QuestLogWrapperProps {
-  questsResults: ReturnType<typeof useGetQuests>
+  quests: Quest[] | null | undefined
+  questIdToQuestStateMap: Record<number, QuestLogInfo['state'] | undefined>
+  isLoading: boolean
   projectId: string
   selectedQuestId: number | null
   setSelectedQuestId: (id: number | null) => void
 }
 
 export function QuestLogWrapper({
-  questsResults,
+  quests,
+  questIdToQuestStateMap,
+  isLoading,
   projectId: appName,
   selectedQuestId,
   setSelectedQuestId
 }: QuestLogWrapperProps) {
-  const quests = questsResults?.data?.data
   const userCredits = useGetG7UserCredits()
   const userCreditsBalance = userCredits?.data?.data
   const pointsBalancesQuery = useGetPointsBalancesForProject(appName)
   const pointsBalances = pointsBalancesQuery?.data?.data
   const { t } = useTranslation()
-  const { questIdToQuestStateMap, isPending: isGetQuestStatesPending } =
-    useGetQuestStates({
-      quests
-    })
```

Comment:
why did you decide to move this hook one level up and remove the quest result from the props? I'm ok with passing quests and isLoading but I'm not sure I understand why we need questIdToQuestStateMap as prop.

We will eventually need to move this component to the quest-ui too as we need these changes in the browser platform too. In that case why don't we keep the pattern of having the hooks inside of the component?

Reply:
If the quest is completed but not claimable, it shouldn't be shown in the UI as the user cannot interact with it any more. We need it in the level up so we can filter it out and select the first quest from the quest array, which affects what we show in quest details.

I've updated the logic we will share with the store here https://github.com/HyperPlay-Gaming/quests-ui/pull/40 with updated tests. In the store, I figured we might SSR it. If not, I think we should refactor `useGetQuestStates` to be shared in quests-ui at that time. We still don't have many (any?) live quests in the store right now, so let's not block this PR on that basis.